src/Container/Types.hs

```diff
@@ -0,0 +1,45 @@
+{-# LANGUAGE DeriveAnyClass #-}
```

Comment:
[nit] These types seem to only be used in `Tarball`, tests aside. Is there a reason they should not be included in that module?

Reply:
Leaving it open for other sources which may not be tarball centric